src/cpu_platform.cpp

```diff
@@ -0,0 +1,41 @@
+#include "cpu_platform.h"
+#include "runtime.h"
+
+#include <algorithm>
+#include <fstream>
+#include <iterator>
+#include <string>
+
+#if defined(__APPLE__)
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#endif
+
+
+CpuPlatform::CpuPlatform(Runtime* runtime)
+    : Platform(runtime)
+{
+    #if defined(__APPLE__)
+    size_t buf_len;
+    sysctlbyname("machdep.cpu.brand_string", nullptr, &buf_len, nullptr, 0);
+    char buf[buf_len];
```

Comment:
Note that this is using VLAs which are available as a GNU extension but not part of standard C++. Maybe better use std::string?